test/e2e/mcd_test.go

```diff
@@ -187,8 +187,8 @@ func TestKernelType(t *testing.T) {
 	assert.Equal(t, infraNode.Annotations[constants.CurrentMachineConfigAnnotationKey], renderedConfig)
 	assert.Equal(t, infraNode.Annotations[constants.MachineConfigDaemonStateAnnotationKey], constants.MachineConfigDaemonStateDone)
 
-	kernelInfo := execCmdOnNode(t, cs, infraNode, "uname", "-a")
-	if !strings.Contains(kernelInfo, "PREEMPT RT") {
+	kernelInfo := execCmdOnNode(t, cs, infraNode, "chroot", "/rootfs", "rpm", "-qa", "kernel-rt-core")
```

Comment:
The `-a` there is unnecessary (but fine as is, we can address it in a followup).  We should also probably make `execCmdOnNode` do the `chroot /rootfs` internally because basically everything else in this function is targeting the real root too.

Reply:
-a is not necessary but `-a` explicitly says `query/verify all packages` and that made me use it :)
yeah, you are right. We can look at execCmdOnNode() to do  `chroot /rootfs`

Reply:
the funny thing is the other uses of `execCmdOnNode` play from outside the chroot by just string matching on files within `/rootfs` - but I agree we can implement chroot in the function to make it transparent that commands are executed on the node's root directly